src/main/java/seedu/address/model/person/Person.java

```diff
@@ -180,14 +180,17 @@ public void updateAttribute(Attribute attribute) {
      * Deletes an attribute from the person.
      *
      * @param attributeType The type of the attribute to delete.
+     * @return
      */
-    public void deleteAttribute(String attributeType) {
-        if (!attributes.containsKey(attributeType)) {
-            throw new IllegalArgumentException("Attribute with name " + attributeType + " does not exist");
+    public boolean deleteAttribute(String attributeType) {
+        if (attributes.containsKey(attributeType)) {
+            attributes.remove(attributeType);
+            return true;
         }
-        attributes.remove(attributeType);
+        return false;
     }
 
+
```

Comment:
Is the condition of attribute to delete a result of an error, or an expected typical outcome? If it is an error, like user input issue, I think its better handled with exceptions.